modules/core/src/main/scala/org/scalasteward/core/util/string.scala

```diff
@@ -70,7 +72,29 @@ object string {
   ): Option[String] = {
     var changed = false
     val replacer1 = replacer.andThen(_.map(r => { changed = true; r }))
-    val result = regex.replaceSomeIn(target, replacer1)
+    val targetString = target.toString
+    val result = if (!targetString.contains("scala-steward:off")) {
+      regex.replaceSomeIn(target, replacer1)
+    } else {
+      var ignoreLines = false
+      targetString.linesIterator
+        .map(s => {
+          if (ignoreLines) {
+            if (s.contains("scala-steward:on")) {
+              ignoreLines = false
+              regex.replaceSomeIn(s, replacer1)
+            } else s
+          } else {
+            if (s.contains("scala-steward:off")) {
+              if (ignoreLines || regexIgnoreMultiLinesBegins.findFirstIn(s).isDefined) {
+                ignoreLines = true
+              }
+              s
+            } else regex.replaceSomeIn(s, replacer1)
+          }
+        })
+        .mkString(System.lineSeparator())
```

Comment:
I think this could lead to huge diffs if the original file didn't use `System.lineSeparator()` as line separator.

Reply:
Changed from `str.lineIterator` to `str.linesWithSeparators` to preserve original end of lines in https://github.com/fthomas/scala-steward/pull/671/commits/d964a63744825d758baa22349c16af87ed2180b7

Reply:
I didn't know about `linesWithSeparators`. Nice!